indexing-service/src/main/java/io/druid/indexing/overlord/TaskRunner.java

```diff
@@ -43,13 +45,18 @@
    *
    * @param taskid task ID to clean up resources for
    */
-  public void shutdown(String taskid);
+  void shutdown(String taskid);
 
-  public Collection<? extends TaskRunnerWorkItem> getRunningTasks();
+  Collection<? extends TaskRunnerWorkItem> getRunningTasks();
 
-  public Collection<? extends TaskRunnerWorkItem> getPendingTasks();
+  Collection<? extends TaskRunnerWorkItem> getPendingTasks();
 
-  public Collection<? extends TaskRunnerWorkItem> getKnownTasks();
+  Collection<? extends TaskRunnerWorkItem> getKnownTasks();
 
-  public Collection<ZkWorker> getWorkers();
+  /**
+   * Some runners are able to scale up and down their capacity in a dynamic manner. This returns stats on those activities
+   *
+   * @return ScalingStats if the runner has an underlying resource which can scale, Optional.absent() otherwise
+   */
+  Optional<ScalingStats> getScalingStats();
```

Comment:
My issue here is how to handle `io.druid.indexing.overlord.http.OverlordResource#getScalingState`

This is intended to be a baby step